cli/config/src/lib.rs

```diff
@@ -60,15 +60,21 @@ impl Config {
     /// Locates and loads in the user's configuration file.  We search for the configuration file
     /// in the following locations, in order:
     ///
+    ///   - Location specified by the path paramter if provided
     ///   - `$TREE_SITTER_DIR/config.json`, if the `TREE_SITTER_DIR` environment variable is set
     ///   - `tree-sitter/config.json` in your default user configuration directory, as determined
     ///     by [`dirs::config_dir`](https://docs.rs/dirs/*/dirs/fn.config_dir.html)
     ///   - `$HOME/.tree-sitter/config.json` as a fallback from where tree-sitter _used_ to store
     ///     its configuration
-    pub fn load() -> Result<Self> {
-        let Some(location) = Self::find_config_file()? else {
-            return Self::initial();
+    pub fn load(path: Option<PathBuf>) -> Result<Self> {
+        let location = match (path, Self::find_config_file()?) {
+            (Some(alt_path), _) => alt_path,
+            (None, Some(default_path)) => default_path,
+            _ => {
+                return Self::initial();
+            }
         };
+
```

Comment:
Just a nitpick, i find it a bit cleaner to maybe do the following instead of the interleaving match

```suggestion
    pub fn load(path: Option<PathBuf>) -> Result<Self> {
        let location = if let Some(path) = path {
            path
        } else if let Some(path) = Self::find_config_file()? {
            path
        } else {
            return Self::initial();
        };
```

Reply:
Makes sense, I went back and forth between the two myself :)

Reply:
yeah I tend to flip flop too, it's a case-by-case basis for me ;)